pages/api/poap.ts

```diff
@@ -0,0 +1,102 @@
+import { env } from "@/config/env.mjs";
+import type { NextApiRequest, NextApiResponse } from "next";
+import {
+  createPublicClient,
+  getAddress,
+  http,
+  isAddress,
+  type Address
+} from "viem";
+import { mainnet, sepolia } from "viem/chains";
+
+export interface PoapResponseType {
+  event: {
+    id: number;
+    fancy_id: string;
+    name: string;
+    event_url: string;
+    image_url: string;
+    country: string;
+    city: string;
+    description: string;
+    year: number;
+    start_date: Date;
+    end_date: Date;
+    expiry_date: Date;
+    tokenId: string;
+    owner: string;
+  };
+}
+
+const POAP_EVENT_ID = env.POAP_EVENT_ID;
+// const POAP_EVENT_ID = 177985; <- use this for testing
+
+const publicClient = createPublicClient({
+  chain: env.NEXT_PUBLIC_IS_MAINNET ? mainnet : sepolia,
+  transport: http()
+});
+
+async function getAddressFromQuery(address: string): Promise<Address> {
+  if (isAddress(address)) {
+    return getAddress(address);
+  }
+
+  // check if its an ENS name
+  const resolvedAddress: Address | null = await publicClient.getEnsAddress({
+    name: address
+  });
+  if (resolvedAddress) {
+    return resolvedAddress;
+  }
+
+  throw new Error("Invalid ethereum address");
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "GET") {
+    try {
+      const { address } = req.query as { address: string };
+      const normalizedAddress = await getAddressFromQuery(address);
+
+      const poapRes = await fetch(
+        `https://api.poap.tech/actions/scan/${normalizedAddress}/${POAP_EVENT_ID}`,
+        {
+          headers: {
+            accept: "application/json",
+            "X-API-KEY": env.POAP_API_KEY
+          },
+          method: "GET"
+        }
+      );
+      if (poapRes.status === 404) {
+        console.info("User does not hold this POAP.");
+        res.status(201).send({
```

Comment:
why do you send a `201` response? `201` is the success message for a resource creation

Reply:
I changed this to a 204 response - which is No Content
is this alright?